cpp/src/arrow/util/thread-pool.cc

```diff
@@ -186,6 +220,7 @@ void ThreadPool::WorkerLoop(std::shared_ptr<State> state,
 
 Status ThreadPool::SpawnReal(std::function<void()> task) {
   {
+    FixIfBroken();
```

Comment:
What is the rough cost of this (e.g. # of nanoseconds)? I guess this can be measured in the benchmarks

Reply:
According to https://stackoverflow.com/a/41783404 it is < 10ns. This seems OK but we should definitely keep an eye on benchmarks for thread spawning overhead